### ISSUE TITLE
shouldn't loss be calculated between decoder_outputs and projected layer of logits?

### ISSUE THREAD

**Issue by rajarsheem** [tensorflow/nmt#55]
In [L328 of model.py](https://github.com/tensorflow/nmt/blob/master/nmt/model.py#L328) , I see that ```BasicDecoder``` is not provided with an argument ```output_layer``` which implies that we are getting the raw rnn outputs while doing ```dynamic_decode``` later. As a result, it seems that we are calculating the loss between ```logits``` which are the actually the rnn_outputs and decoder labels. Is this the intended behavior or am I missing something?

**Comment by rajarsheem** [comment]
Oh, sorry, I missed [this](https://github.com/tensorflow/nmt/blob/master/nmt/model.py#L350) line.
